Tests/SonarQube.Bootstrapper.Tests/BuildAgentUpdaterTests.cs

```diff
@@ -68,6 +70,96 @@ public void Updater_CheckDownloadUrl_ConnectionFailure()
             CheckInvalidUrlFails("http://localhost:9000");
         }
 
+        [TestMethod]
+        [Description("The targets file should be copied if none are present. The files should not be copied if they already exist and have not been changed.")]
+        public void InjectTargetsFile_Copy()
+        {
+            // In case the dummy targets file somehow does not get deleted (e.g. when debugging) , make sure its content is valid XML 
+            string sourceTargetsContent = @"<Project ToolsVersion=""4.0"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"" />";
+            CreateDummySourceTargetsFile(sourceTargetsContent);
+
+            try
+            {
+                InjectTargetsFileAndAssert(sourceTargetsContent, expectCopy: true);
+
+                // if we try to inject again, the targets should not be copied because they have the same content
+                InjectTargetsFileAndAssert(sourceTargetsContent, expectCopy: false);
+            }
+            finally
+            {
+                foreach (string destinationDir in BuildAgentUpdater.DestinationDirs)
```

Comment:
It would be safer to clean up the directories at the start of the test too (in case of an aborted test run, or in case the user has manually run the bootstrapper on their machine previously).
You've also repeated this code in the next test - factor it out into a separate method.